renderdoc/driver/d3d12/d3d12_pixelhistory.cpp

```diff
@@ -638,15 +638,16 @@ struct D3D12PixelHistoryCallback : public D3D12ActionCallback
       // Copy into a buffer, but treat the footprint as the same format as the target image
       uint32_t elementSize = GetByteSize(0, 0, 0, p.copyFormat, 0);
 
+      UINT bufferElementWidth = (UINT)m_CallbackInfo.dstBuffer->GetDesc().Width / elementSize;
       dst.Type = D3D12_TEXTURE_COPY_TYPE_PLACED_FOOTPRINT;
       dst.pResource = m_CallbackInfo.dstBuffer;
       dst.PlacedFootprint.Offset = 0;
-      dst.PlacedFootprint.Footprint.Width =
-          (UINT)m_CallbackInfo.dstBuffer->GetDesc().Width / elementSize;
-      dst.PlacedFootprint.Footprint.Height = 1;
+      dst.PlacedFootprint.Footprint.Width = RDCMIN(bufferElementWidth, 16384U);
+      dst.PlacedFootprint.Footprint.Height =
+          RDCMAX(1U, bufferElementWidth / dst.PlacedFootprint.Footprint.Width);
```

Comment:
Rather than adding complexity with the math here to get x/y dimensions and offsets, could the code instead store an array of buffers, which are allocated on demand as we need to copy more pixels for events? It seems that the solution in this PR will run into the same issue as the previous code, just with a higher threshold before failing.

Reply:
Good point about failing if Height > 16384. I think we are ok a 16384 x 16384 texture with 4-bytes per pixel has enough space for a buffer of millions of history events. I think other bits of code will fall over if we get to a million history events.

To me this is a GPU memcpy and we are finding the correct parameters to make the known destination byte offset be correct for the D3D12 API. I think this approach is simpler than having an array of buffer resources and indexing into (that is basically manually doing a 2D array of bytes i.e. a Texture).

I will put a check in for Height being too large then increasing the Depth.